delocate/delocating.py

```diff
@@ -80,29 +91,51 @@ def delocate_tree_libs(lib_dict, lib_path, root_path):
             if not exists(required):
                 raise DelocationError('library "{0}" does not exist'.format(
                     required))
-            copied_libs[required] = requirings
+            # Copy requirings to preserve it since it will be modified later.
+            copied_libs[required] = requirings.copy()
             copied_basenames.add(r_ed_base)
         else:  # Is local, plan to set relative loader_path
             delocated_libs.add(required)
-    # Modify in place now that we've checked for errors
-    for required in copied_libs:
-        shutil.copy(required, lib_path)
-        # Set rpath and install names for this copied library
-        for requiring, orig_install_name in lib_dict[required].items():
-            req_rel = relpath(rp_lib_path, dirname(requiring))
-            set_install_name(requiring, orig_install_name,
-                             '@loader_path/{0}/{1}'.format(
-                                 req_rel, basename(required)))
+    # Modify in place now that we've checked for errors.
+    # Copy libraries outside of root_path to lib_path.
+    for old_path in copied_libs:
+        new_path = realpath(pjoin(lib_path, basename(old_path)))
+        logger.info(
+            "Copying library %s to %s", old_path, relpath(new_path, root_path)
+        )
+        shutil.copy(old_path, new_path)
+        # Delocate this file now that it is stored locally.
+        delocated_libs.add(new_path)
+        # Update lib_dict with the new file paths.
```

Comment:
Now modifying `lib_dict` in-place.  Can we fill another dictionary instead?  Or copy `lib_dict` to avoid modifying it in-place?   Not that anyone would likely notice, but just to keep the function behavior the same.

Reply:
I don't think I realized I modified it in-place.

I'll switch the `Dict` types to a `Mapping` type which will make the input parameter read-only, then copy the dictionary before modifying it.